Protocol.md

```diff
@@ -23,188 +44,182 @@ The following figure gives a brief overview of the protocol:
+The sending node SHOULD:
 
----
+  - set `feerate_per_vb` to at least the rate it estimates would cause the transaction to be immediately included in a block.
+  - set `contract_maturity_cltv` to the earliest expected oracle signature time.
+  - set `contract_timeout_cltv` sufficiently long after `contract_maturity_cltv` to allow for late oracle signatures and other delays to closing the contract.
 
-### Accept
-After receiving the `offer` message from A, and after [validating](#offer-validation) the provided information, B creates and sends back an `accept` message containing:
-1. B's fund amount (satoshi)
-1. B's public keys
-1. B's addresses
-1. B's inputs
-1. CET signatures
-1. Refund signature
+The receiving node MUST:
 
-#### B's fund amount (satoshi)
-How much B inputs into the contract.
+  - ignore undefined bits in `channel_flags`.
 
-#### B's public keys
-The following public keys:
+The receiving node MAY reject the contract if:
 
-1. Funding Public Key
-   - B's public key used in the multi-signature [funding output](Transactions.md#FundingOutputs)
-2. Sweep Public Key
-   - B's public key used in the creation of the CETs [output script](Transactions.md#CETOutputs).
+  - it does not agree to the terms in `contract_info`.
+  - the `contract_info` is missing relevant events.
+  - it does not want to use the oracle(s) specified in `oracle_info`.
+  - `total_collateral_satoshis` is too small.
 
-#### B's addresses
+The receiving node MUST reject the contract if:
 
-The following addresses:
+  - the `chain_hash` value is set to a hash of a chain that is unknown to the receiver.
+  - the `contract_info` refers to events unknown to the receiver.
+  - the `oracle_info` refers to an oracle unknown to the receiver.
+  - it considers `feerate_per_vb` too small for timely processing or unreasonably large.
+  - `funding_pubkey` is not a valid secp256k1 pubkey in compressed format.
+  - the funder's amount for the funding transaction is not sufficient for full [fee payment](Transactions.md#fee-payment).
 
-1. Funding Change Address
-   - The address to use to send the change if the sum of A's inputs is greater than the fund amount plus the fees.
-2. Final Address
-   - The address to which funds will be sent in [unilateral contract execution](Transactions.md#CETOutputs), [refund](Transactions.md#refund-transaction) and [mutual closing](Transactions.md#mutual-closing-transaction) transactions.
+### The `accept_dlc` Message
 
-#### B's inputs
-The set of UTXOs to be used as input to the fund transaction.
+This message contains information about a node and indicates its
+acceptance of the new DLC, as well as its CET and refund transaction
+signatures. This is the second step toward creating the funding transaction
+and closing transactions.
 
-#### CET signatures
-A set of signatures from B, one for each CET input.
+1. type: ??? (`accept_dlc`)
+2. data:
+   * [`32*byte`:`temporary_contract_id`]
+   * [`u64`:`total_collateral_satoshis`]
```

Comment:
Isn't that implicit from the offer message? I would say it should be equal to `outcome amount - offerer collateral`?

Reply:
My only concern with that is a situation where either party would intentionally want to over-collateralize (say to improve privacy in their lose cases) in which case it doesn't hurt to have this around and just use this rule you mention as the default verification?

Reply:
I disagree, I don't perceive any real addition to complexity, there's no reason not to include this in the message if there are use cases that have this determined implicitly and those that don't. Making it explicit is not an added complexity.

Reply:
Ok I think I got what you mean and why you put it. Though how does that affect the contract outcomes then? Let's say offerer sent contract with collaterals: 1 for offerer and 1 for receiver. Outcomes are all 2. But receiver would like to put 2 as collateral for privacy as you mention. Should offerer assume that the outcomes are now 3 when constructing the CETs? If so I guess it should be specified explicitely.

Reply:
I plan to solve this by having `contract_info` be a one-sided POV data structure, i.e. contains the payouts to the offerer, thus the remainder of the funds (total_funding - outcome) goes to the accepter where total_funding is the sum of total_collaterals.

Thus, in the case you specified the offerer has a contract info with outcomes of two, and posts a total_collateral of 1. Now the accepter can choose either to post 1 and have things be like the offerer may be expecting or they are allowed to post 2, in which case their outputs will be either 1 or 3.

Reply:
I think what you wrote above should be written in an issue so we remember it.

Reply:
Tracked here: https://github.com/discreetlogcontracts/dlcspecs/issues/62